pkg/environment/types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package environment
+
+import (
+	"context"
+	"testing"
+)
+
+type Operation func (context.Context, Environment)(context.Context, error)
+
+type Environment interface {
+	// AddPrereq adds an environment operation to be executed
+	// prior to the environment being ready
+	AddPrereq(op Operation) Environment
```

Comment:
🤔 is `AddPrereq` or `AddPreCheck` or `AddSetup` more accurate? If it's _to be executed_ then I'm leaning to use `AddSetup`. In my mind, `AddPrereq` sounds like "will check and error out if it's not satisfied".